transitions/registration.js

```diff
@@ -310,20 +311,18 @@ module.exports = {
         var doc = options.doc,
             db = options.db,
             audit = options.audit,
-            patientId = doc.patient_id,
+            patientShortcode = doc.patient_id,
             patientNameField = _.first(options.params) || 'patient_name';
 
-        utils.getRegistrations({
-            db: db,
-            id: patientId
-        }, function(err, registrations) {
-            if (err) {
+        utils.getPatientContactUuid(db, patientShortcode, function(err, patientUuid) {
+            if (err && err.statusCode !== 404) {
                 return callback(err);
             }
-            if (registrations.length) {
-                // patient already registered, no action required
+
+            if (patientUuid) {
                 return callback();
```

Comment:
Could use a comment here - maybe you could reuse the one from 2 lines above that you deleted? :P